src/replication.c

```diff
@@ -808,6 +761,27 @@ update_replica_entry(spec_t *spec, replica_t *replica, int iofd, char *replicaip
 	free(rio);
 	rio = NULL;
 	update_volstate(spec);
+	struct epoll_event event;
```

Comment:
Even though not related to this code change, please move addition of replica to spec->rq and update_volstate towards end of the function.
So, first all the housekeeping work like creating threads etc, then, adding replica to spec->rq and then update_volstate.